scene/resources/animation.cpp

```diff
@@ -5677,7 +5677,7 @@ Variant Animation::interpolate_variant(const Variant &a, const Variant &b, float
 		}
 		case Variant::INT: {
 			const int64_t va = a.operator int64_t();
-			return int(va + ((b.operator int64_t()) - va) * c);
+			return int64_t(va + ((b.operator int64_t()) - va) * c);
 		}
 		case Variant::FLOAT: {
 			const real_t va = a.operator real_t();
```

Comment:
Related issue: This should be `double`, not `real_t`. `Variant::FLOAT` is always `double`, but it's a common misunderstanding from contributors that it should be `real_t`.

`real_t` is only used for vector types (Vector2*, Transform, etc.).

Reply:
The whole file seems to use `real_t` for scalars. It could be that it's intentional, or a bug. Maybe best left for a separate issue/PR.

Reply:
Will take a look!

Reply:
Also found another quirk with the size iteration, fixed